src/drive/web/modules/filelist/FileIconShortcut.jsx

```diff
@@ -8,20 +8,22 @@ import GlobeIcon from 'cozy-ui/transpiled/react/Icons/Globe'
 const FileIconShortcut = ({ file, size = 32 }) => {
   const client = useClient()
   const { shortcutImg } = useFetchShortcut(client, file.id)
-  return (
-    <>
-      <div style={{ display: shortcutImg ? 'block' : 'none' }}>
-        <img src={shortcutImg} width={size} height={size} />
-      </div>
-      <div
-        style={{
-          display: !shortcutImg ? 'block' : 'none'
+  const [isBroken, setBroken] = useState(null)
+
+  if (shortcutImg && !isBroken) {
+    return (
+      <img
+        src={shortcutImg}
+        width={size}
+        height={size}
+        onError={() => {
+          setBroken(true)
         }}
```

Comment:
why did you remove all the `div style`? Are you sure that was not needed? 

How an image coming from the registry can be broken?

Reply:
I removed the `div style` during my tests and left it in my final version because it was consistent with the way react works. Did I miss an optimisation to avoid reloading the image?

I don't know, I was inspired by the way AppIcon works in cozy-ui (cf: [code](https://github.com/cozy/cozy-ui/blob/6166abd6346fbfb77da8f058124e65340410282c/react/AppIcon/index.jsx#L114)).

Reply:
"how react works" I was saying more declarative. I don't think I changed the display. If there is an image it displays it else I display the fallback 🤔 From my point of view, the change is that the component always return what it displays without manipulating it with CSS